src/main/kotlin/uk/gov/dluhc/printapi/messaging/ProcessPrintResponseFileMessageListener.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.dluhc.printapi.messaging
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.awspring.cloud.messaging.listener.annotation.SqsListener
+import mu.KotlinLogging
+import org.springframework.messaging.handler.annotation.Payload
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseFileMessage
+import uk.gov.dluhc.printapi.printprovider.models.PrintResponses
+import uk.gov.dluhc.printapi.service.SftpService
+import javax.validation.Valid
+
+private val logger = KotlinLogging.logger { }
+
+/**
+ * Implementation of [MessageListener] to handle [ProcessPrintResponseFileMessage] messages
+ */
+@Component
+class ProcessPrintResponseFileMessageListener(
+    private val sftpService: SftpService,
+    private val objectMapper: ObjectMapper,
+) : MessageListener<ProcessPrintResponseFileMessage> {
+
+    @SqsListener("\${sqs.process-print-response-file-queue-name}")
+    override fun handleMessage(@Valid @Payload payload: ProcessPrintResponseFileMessage) {
+        with(payload) {
+            val filePath = "$directory/$fileName"
+            logger.info { "Begin processing PrintResponse file [$filePath]" }
+            val printResponsesString = sftpService.fetchFile(filePath)
+            val printResponses = objectMapper.readValue(printResponsesString, PrintResponses::class.java)
+            processPrintResponses(printResponses)
+            sftpService.removeFileFromOutBoundDirectory(filePath)
+            logger.info { "Completed processing PrintResponse file [$filePath]" }
```

Comment:
I would have expected a service class to be responsible for managing these details and listener only calls that service method.

Service should handle all exceptions that should not cause the SQS message to be retried.

Reply:
I'd vote for that - sounds like a good idea.

Reply:
I've abstracted this functionality into uk.gov.dluhc.printapi.service.PrintResponseFileService#processPrintResponseFile 👍